### PR TITLE
add Stride Diagnostic Analyzer blog post

### PR DESCRIPTION
**Pull request by IXLLEGACYIXL**
This adds a blog post about the upcomming feature for Stride.Core.CompilerServices DiagnosticAnalyzers

A review about grammar and typos is needed as always :D

### CODE REVIEW

**Comment by VaclavElias** [comment]
Let me know once you are ready, and I can publish it to staging for further review :)

**Comment by IXLLEGACYIXL** [comment]
im ready, i just failed again with VSCode to rollback the commit .. i should have used revert and not reset.. but it is what it is

File: posts/2023-10-23-new-diagnostic-analyzers-feature.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+## What is a DiagnosticAnalyzer?
+
+A `DiagnosticAnalyzer` is a Roslyn feature to scan your code while you are typing in your IDE.
+C# uses them too, everytime you see a warning or red squiggly lines in your IDE, a `DiagnosticAanalyzer` get's triggered.
+This analyzation has `NOTHING` to do with telemetry.
+
+## What does it do?
+
+The new `Diagnostics` cover mostly `Serialization`, the analyzers will create warnings when code is written that is incompatible with [Strides Serialization Rules of Thumb](https://github.com/stride3d/stride-docs/blob/master/en/manual/scripts/serialization.md#rule-of-thumb).
```
Comment (VaclavElias) [inline posts/2023-10-23-new-diagnostic-analyzers-feature.md:28]: Did you want to link actually to this location 

https://doc.stride3d.net/latest/en/manual/scripts/serialization.html

**Review by VaclavElias — COMMENTED** [review]
Added a comment

File: posts/2023-10-23-new-diagnostic-analyzers-feature.md (hunk 1 above)
Comment (VaclavElias) [reply posts/2023-10-23-new-diagnostic-analyzers-feature.md]: No problem. And obviously, I remember as I mess around docs 🤣

**Comment by VaclavElias** [comment]
Also, add yourself here to the authors section https://github.com/stride3d/stride-website/blob/master/_data/site.json

File: posts/2023-10-23-new-diagnostic-analyzers-feature.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+## What is a DiagnosticAnalyzer?
+
+A `DiagnosticAnalyzer` is a Roslyn feature to scan your code while you are typing in your IDE.
+C# uses them too, everytime you see a warning or red squiggly lines in your IDE, a `DiagnosticAanalyzer` get's triggered.
+This analyzation has `NOTHING` to do with telemetry.
+
+## What does it do?
+
+The new `Diagnostics` cover mostly [Serialization](https://doc.stride3d.net/latest/en/manual/scripts/serialization.html), the analyzers will create warnings when code is written that is incompatible with [Strides Serialization Rules of Thumb](https://github.com/stride3d/stride-docs/blob/master/en/manual/scripts/serialization.md#rule-of-thumb).
```
Comment (VaclavElias) [inline posts/2023-10-23-new-diagnostic-analyzers-feature.md:28]: Actually, what I meant, that your reference to the **Rules of Thumb** is here https://doc.stride3d.net/latest/en/manual/scripts/serialization.html#rule-of-thumb

Comment (IXLLEGACYIXL) [reply posts/2023-10-23-new-diagnostic-analyzers-feature.md:28]: that was intentional as the rule of thumb is not accurate currently and my diagnostics arent either

**Pull request merged**